lib/passpartu/check_policy.rb

```diff
@@ -0,0 +1,59 @@
+module Passpartu
+  class CheckPolicy
+    CRUD_KEY = 'crud'.freeze
+
+    def initialize(keys)
+      @hash = Passpartu.policy
+      @keys = keys
+      @waterfall_rules = Passpartu.config.waterfall_rules
+    end
+
+    def self.call(keys)
+      new(keys).call
+    end
+
+    def call
+      check_policy
+      check_crud if policy_missed? && last_key_crud?
+      result
+    end
+
+    private
+
+    attr_reader :hash, :keys, :waterfall_rules, :result
+
+    def check_policy
+      loop_hash = hash.dup
+      keys.each_with_index do |key, index|
+        if loop_hash[key].is_a? Hash
+          loop_hash = loop_hash[key]
+          next
+        elsif waterfall_rules || last?(index)
```

Comment:
check this config outside method logic. Otherwords - do not check waterfall if it's turned off